File: Entities/AppAccounts/dbAccounts.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ALBAB.Entities.AppAccounts
{
        public class dbAccounts
    {
        public int Id { get; set; }

       [Required]
        public string KeyId { get; set; }

        [Required]
        public string Name { get; set; }


        public int lvl { get; set; }

        public DateTime Created { get; set; }   = DateTime.Now;

        public dbAccounts Parent { get; set; }

       // Reference for self
      
        public int? ParentId {get;set;}
        public ICollection <dbAccounts> Children { get; set; }

        public bool IsExpandable {get;set;} = false;

      /*   public dbAccounts(){

            Children = new Collection<dbAccounts>();

        } */

    }


}

File: ClientApp/src/app/models/dbaccounts.ts
export interface dbAccounts {
  id: number;
  keyId: string;
  name: string;
  lvl: number;
  created: string;
  parent: dbAccounts;
  parentId: number | null;
  isExpandable: boolean;

}
export interface dbAccountsNode {
  id: number;
  keyId: string;
  name: string;
  lvl: number;
  created: string;
  parent: dbAccounts;
  parentId: number | null;
  isExpandable: boolean;
  selected?: boolean;
  children: dbAccountsNode[];

}
export class dbAccountsNewChild {
  id:number
  keyId: string;
  name: string;
  lvl: number;
  parentId: number;
  isExpandable: boolean;
}

